File: fixtures/kotlin/type_inference/lambda/lambda.kt
val toLabel: (Int) -> String = { n -> "n=$n" }

fun applyTwice(value: Int, fn: (Int) -> Int): Int = fn(fn(value))

fun main() {
    val doubled = applyTwice(21) { x -> x * 2 }
    val lengths = listOf("a", "bb", "ccc").map { it.length }
    println("$toLabel $doubled $lengths")
}

File: fixtures/typescript/type_inference/lambda/lambda.ts
type Mapper<T, U> = (value: T) => U;

const toLabel: Mapper<number, string> = (n) => `n=${n}`;

function applyTwice(value: number, fn: (x: number) => number): number {
    return fn(fn(value));
}

function fetchGreeting(name: string): Promise<string> {
    return Promise.resolve(`hello ${name}`);
}

const doubled = applyTwice(21, (x) => x * 2);
const chained = fetchGreeting("ada").then((greeting) => greeting.length);

export { toLabel, applyTwice, fetchGreeting, doubled, chained };
